eth/downloader/downloader.go

```diff
@@ -798,6 +799,26 @@ func (d *Downloader) findAncestor(p *peerConnection, remoteHeader *types.Header)
 		}
 	}
 
+	ancestor, err := d.findAncestorSpanSearch(p, mode, remoteHeight, localHeight, floor)
+	if err == nil {
+		return ancestor, nil
+	}
+	// The returned error was not nil.
+	// If the error returned does not reflect that a common ancestor was not found, return it.
+	// If the error reflects that a common ancestor was not found, continue to binary search,
+	// where the error value will be reassigned.
+	if err != errNoAncestorFound {
```

Comment:
We've started using this format, instead of checking explicit equality:
```suggestion
	if !errors.Is(err,errNoAncestorFound) {
```